File: DSA.SPA/ClientApp/pages/rent/api/rent-service.js
import * as httpService from '../../../api/http-service'

const apiRent = "/api/rent/";

export const getRentsByParams = data => {
    let params = {
        url: apiRent + "getRentsByParams",
        data: data
    }

    return httpService.postData(params);
}

export const addRent = data => {
    let params = {
        url: apiRent + "addRent",
        data: data
    }

    return httpService.postData(params);
}

File: SAT.IoT/Service/RentService.cs
using SAT.BLL.Dto.Rents;
using SAT.Domain.Params;
using SAT.IoT.Extenction;

namespace SAT.IoT.Service
{
    public class RentService
    {
        private const string _baseRentUrl = "/api/rent/";
        private const string _updateRent = "updateRent";
        private const string _getRents = "getRents";

        public static void UpdateRent(RentDto updateRent)
        {
            HttpClientExtenction.PostData(updateRent, string.Concat(_baseRentUrl, _updateRent));
        }

        public static CollectionResult<RentDto> GetRents(RentsFilterParams filterParams)
        {
            var result = HttpClientExtenction.PostDataAndGetResult<RentsFilterParams, CollectionResult<RentDto>>(filterParams, string.Concat(_baseRentUrl, _getRents));
            return result;
        }
    }
}
